heisenbridge/channel_room.py

```diff
@@ -140,6 +140,13 @@ def init(self) -> None:
         )
         self.commands.register(cmd, self.cmd_stop, ["STOP!", "STAHP", "STAHP!"])
 
+        cmd = CommandParser(
```

Comment:
I'd call this `UPGRADE` which would do the following:

if room version <9 AND the HS supports it, create new room (min v9), copy all user configurable room data (name, avatar etc.), invite the user, invalidate old room and make it gc'd, doing a real room upgrade doesn't really work with heisenbridge as it does not intentionally keep the rooms that are not actively used and a tombstoned history room wouldn't fit in that category
if room version >=9, just add join rules

I'm also fine if it checks that the room version is <9 and then says you need to leave the room and join the channel again to upgrade manually.

This command can be "slow" in the sense it fetches and checks things as it's not a hot path.

Reply:
I'm not a huge fan of performing that kind of complex upgrade in the bridge. If a user wants it, most clients can do it.
It's also apparently not easily possible to check if a room version is >= something, since the version is not neccesarily numeric.
So basically, it'd have to be a check if the room value is not any of versions 1 to 8, which is a bit ugly.

So currently the approach is to try and add the join_rule, and if it fails, tell the user to upgrade/re-join the room. Though I'm not sure if adding it will really fail?

Reply:
Clients can't do it as they don't have the required PL. It's also something that wouldn't work without bridge support since upgrading always creates a new room _anyway_ so the bridge needs to move its operations there.

Just check if the room is of unsupported version `in [1,2,3,4,5,6,7,8]` or something and then tell them to leave it if you don't want to try adding support for real upgrades.